plugins/UM3NetworkPrinting/src/SendMaterialJob.py

```diff
@@ -1,99 +1,129 @@
                 with open(signature_file_path, "rb") as f:
-                    parts.append(self.device._createFormPart("name=\"signature_file\"; filename=\"{file_name}\"".format(file_name = signature_file_name), f.read()))
+                    parts.append(self.device._createFormPart(
+                        "name=\"signature_file\"; filename=\"{file_name}\"".format(file_name=signature_file_name),
+                        f.read()))
 
-            Logger.log("d", "Syncing material {material_id} with cluster.".format(material_id = material_id))
-            self.device.postFormWithParts(target = "materials/", parts = parts, on_finished = self.sendingFinished)
+            Logger.log("d", "Syncing material {material_id} with cluster.".format(material_id=material_id))
+            self.device.postFormWithParts(target="materials/", parts=parts, on_finished=self.sendingFinished)
 
     def sendingFinished(self, reply: QNetworkReply):
         if reply.attribute(QNetworkRequest.HttpStatusCodeAttribute) != 200:
-            Logger.log("e", "Received error code from printer when syncing material: {code}".format(code = reply.attribute(QNetworkRequest.HttpStatusCodeAttribute)))
-            Logger.log("e", reply.readAll().data().decode("utf-8"))
\ No newline at end of file
+            Logger.log("e", "Received error code from printer when syncing material: {code}".format(
+                code=reply.attribute(QNetworkRequest.HttpStatusCodeAttribute)))
+            Logger.log("e", reply.readAll().data().decode("utf-8"))
+
+    ##  Parse the reply from the printer
+    #
+    #   Parses the reply to a "/materials" request to the printer
+    #
+    #   \return a dictionary of ClustMaterial objects by GUID
+    #   \throw json.JSONDecodeError Raised when the reply does not contain a valid json string
+    #   \throw KeyErrror Raised when on of the materials does not include a valid guid
+    @classmethod
+    def _parseReply(cls, reply: QNetworkReply) -> Dict[str, ClusterMaterial]:
+        remote_materials_list = json.loads(reply.readAll().data().decode("utf-8"))
+        return {material["guid"]: ClusterMaterial(**material) for material in remote_materials_list}
+
+    ##  Retrieves a list of local materials
+    #
+    #   Only the new newest version of the local materials is returned
+    #
+    #   \return a dictionary of LocalMaterial objects by GUID
+    @classmethod
+    def _getLocalMaterials(cls):
+        result = {}
+        for material in ContainerRegistry.getInstance().findContainersMetadata(type="material"):
+            try:
+                localMaterial = LocalMaterial(**material)
+
+                if localMaterial.GUID not in result or localMaterial.version > result.get(localMaterial.GUID).version:
+                    result[localMaterial.GUID] = localMaterial
+            except (ValueError):
+                Logger.log("e", "Material {material_id} has invalid version number {number}.".format(
```

Comment:
You can use Logger.logException("e") if you want to log the exception as well as log a message.